src/EventListener/ExceptionListener.php

```diff
@@ -16,6 +18,28 @@ public function onKernelException(ExceptionEvent $event): void
         $exception = $event->getThrowable();
         if ($exception instanceof OutOfRangeException) {
             $event->setThrowable(new NotFoundHttpException('Not Found.', $exception));
+        } elseif ($exception instanceof UnexpectedValueException && self::isInternalException($exception)) {
+            $event->setThrowable(new NotFoundHttpException('Not Found', $exception));
         }
     }
+
+    private static function isInternalException(UnexpectedValueException $exception): bool
+    {
+        $messages = [
+            '/^Cannot filter by\: \[.+\] this field is not in allow list$/',
+            '/^Cannot sort by\: \[.+\] this field is not in allow list\.$/',
+            '/^Cannot sort with array parameter$/',
+            '/^ODM query must be a FIND type query$/',
+            '/^There is no component aliased by \[.+\] in the given Query$/',
+            '/^There is no component field \[.+\] in the given Query$/',
+            '/^There is no such field \[.+\] in the given Query component, aliased by \[.+\]$/',
+        ];
+        foreach ($messages as $regex) {
+            if (preg_match($regex, $exception->getMessage()) > 0) {
+                return true;
+            }
+        }
+
+        return false;
+    }
```

Comment:
I'm usually not so fond of matching an exception by its message, especially if we control those exceptions.
Can't we create instead a custom exception (that's extending the `UnexpectedValueException`) and catch that instead of matching the messages?

Reply:
I agree and I already opened an issue for that https://github.com/KnpLabs/knp-components/issues/317
The problem is that we can only add the internal exception in a minor release of knp-components (e.g. 4.1.0) and so the fix in this bundle would stay for the users still using version 4.0.
My idea about the roadmap is: 
1) add the internal exception in the library, release a minor version of the library
2) add support in the bundle for both (new internal exception and ugly old message-based exceptions), release a patch version of the bundle
3) release a new minor version of the bundle, requiring library ^4.1 and removing the ugly old message-based exceptions

Reply:
Great! Sorry, I missed that issue.